Sprint-3/todo-list/script.js

```diff
@@ -12,14 +45,92 @@ let todos = [
 
 populateTodoList(todos);
 
+function toggleComplete(index) {
+  // function to toggle complete status
+  todos[index].completed = !todos[index].completed;
+  populateTodoList(todos);
+}
+
+function deleteTodo(index) {
+  //function to delete specific todo
+  todos.splice(index, 1);
+  populateTodoList(todos);
+}
+populateTodoList(todos); 
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+    let input =
+    document.getElementById("new-todo-input") ||
+    document.getElementById("todo-input") ||
+    document.querySelector('input[type="text"]');
+
+  let todoText = input.value.trim();
+
+  if (todoText === "") {
+    alert("Please enter a todo item!");
+    return;
+  }
+  let newTodo = {
+    // create new todo object
+    task: todoText,
+    completed: false,
+  };
+
+  todos.push(newTodo); // ad todos array
+
+  input.value = ""; //clear input field
+
+  populateTodoList(todos); //refresh the todo list
 }
 
+
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
+  todos = todos.filter((todo) => !todo.completed); // filter out the completed todos
+
+  populateTodoList(todos); // refresh the display
+
+  console.log("The completed todos are deleted!");
 }
+// Event listener
+document.addEventListener("DOMContentLoaded", function () {
+  // Try to find the form by different selectors
+  let form =
+    document.querySelector("form") ||
+    document.querySelector(".add-todo-form") ||
+    document.getElementById("todo-form");
+
+  if (form) {
+    form.addEventListener("submit", addNewTodo);
+  } else {
```

Comment:
🤔 I think the console log can be removed here as it won't serve a purpose to the user since they won't be able to act upon your console log message. 
(Your console log is a good idea once an application is in a Production environment and logging is needed to troubleshoot application problems.)